app/jobs/application_job.rb

```diff
@@ -1,7 +1,3 @@
 class ApplicationJob < ActiveJob::Base
-  # Automatically retry jobs that encountered a deadlock
-  # retry_on ActiveRecord::Deadlocked
-
-  # Most jobs are safe to ignore if the underlying records are no longer available
-  # discard_on ActiveJob::DeserializationError
+  discard_on ActiveJob::DeserializationError # most likely a record that's not there anymore
```

Comment:
it could be nice to flag these errors rather than silently discarding them.

because usually these show up because of an `after_create` rather than `after_create_commit`

Reply:
> usually these show up because of an after_create rather than after_create_commit

What do you mean by this?

Reply:
queueing a job in an `after_create` callback can cause `ActiveJob::DeserializationError` errors because the record has not been saved to the database yet.

https://stackoverflow.com/questions/60224142/activejobdeserializationerror-in-a-after-create-callback-record-not-found

Reply:
Why would you try to do something with a record that's not there yet?

Reply:
to sum things up, there are cases where `after_create` should be changed to `after_create_commit`. 

discarding `ActiveJob::DeserializationError` will cause these incorrect `after_create`s to unnoticed and for the jobs to be silently dropped.

Reply:
another option is to retry these errors twice, and then discard

Reply:
If a record is not saved, and it should be, using a bang method will raise an error essentially saying the same thing. I'm pretty sure you can override it on a case-by-case basis for more critical jobs.